CMSgov/mint-app
# Transaction Helper Functions

# EASI-3507

## Changes and Description

- Implement a WithFunction to Wrap Transactions
- Create an interface to be able to pass either a store or a transaction.
   - Implement these for create Model Plan



## How to test this change
1. Create a model plan
2. Verify that there aren't any errors


Code to pay close attention to
1. [With Transactions](https://github.com/CMSgov/mint-app/compare/main...EASI-3507/db_transaction_helpers_withFunction#diff-92b79a5814d551bc30fd30f4e54e699ff2eff47d5f153be09e1180092e952a0f)
   a. and related test file
      i. Note, this could also go in the resolvers package so it has db clean methods etc.
2. [New Model Plan Resolver](pkg/graph/resolvers/model_plan.go)
3. [Named Preparer interface](pkg/storage/named_preparer.go)


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
